OpenSim/Common/Component.h

```diff
@@ -1554,6 +1554,7 @@ template <class T> friend class ComponentMeasure;
             _connectorsTable[connector.getName()] = ix;
         }
     }
+protected:
```

Comment:
Would prefer to add this to the protected section with the other service methods. I am also thinking we could rename the method to `createSubcomponentsTraversalTree()`.  Also add doxygen comments especially if new "environment" or "study" like Components should call this method as part of the their `extendConnect()` implementation (e.g. TheWorld component in testComponentInterface()).